File: 03b_serialization_in_c#/serialize_in_c#/employee.cs
namespace MyApp.Models
{

    public class Employee
    {
        public String id { get; set; }
        public String name { get; set; } = "unknown";
        double hoursPerWeek { get; set; } = 40;
        double salary = 1234;
        public List<string> languages { get; set; } = [];

        // next line will leed to exception on deserialize
        //public Employee(string s) { this.id = s; }
        public Employee(string id) { this.id = id; }

    }

    public class Person
    {
        public int Id { get; set; }
        public string Name { get; set; } = string.Empty;
        public DateTime BirthDate { get; set; }
    }

}

File: 03_serialization/bin/employee.dart
class Employee {
  Employee(this.id);
  String id;
  String name = "unknown";
  double hoursPerWeek = 40;
  double salary = 1234;
  List<String> languages = [];

  // Object -> Map
  Map<String, dynamic> toJson() {
    return {
      "id": id,
      "name": name,
      "hoursPerWeek": hoursPerWeek,
      "salary": salary,
      "languages": languages,
    };
  }

  // Map -> Object
  factory Employee.fromJson(Map<String, dynamic> json) {
    if (json["id"] == null) {
      throw Exception("no field 'id' found in $json");
    }
    var result = Employee(json["id"]);
    result.name = json['name'];
    result.hoursPerWeek = json['hoursPerWeek'];
    result.salary = json['salary'];
    print(json['languages'].runtimeType); // List<dynamic>
    result.languages = List<String>.from(json['languages']);
    return result;
  }
}
